client-spark/common/src/main/java/org/apache/spark/shuffle/writer/WriteBufferManager.java

```diff
@@ -162,9 +162,10 @@ public WriteBufferManager(
   }
 
   /** add serialized columnar data directly when integrate with gluten */
-  public List<ShuffleBlockInfo> addPartitionData(int partitionId, byte[] serializedData) {
+  public List<ShuffleBlockInfo> addPartitionData(
+      int partitionId, byte[] serializedData, int serializedDataLength) {
```

Comment:
Currently gluten reuse this `serializedData` byte array, only if `serializedDataLength` may be shorter than length of `serializedData`

Reply:
Ok, remove this and change the title